assets/scripts/routes/after-oauth-login.js

```diff
@@ -1,69 +1,68 @@
 // @ts-check
 
 import page from 'page'
+import remember from 'remember'
 
 import {
   svelteTarget,
   defaultRepositoryName,
-  ACCESS_TOKEN_STORAGE_KEY,
   OAUTH_PROVIDER_STORAGE_KEY,
   TOCTOCTOC_ACCESS_TOKEN_URL_PARAMETER,
   TOCTOCTOC_OAUTH_PROVIDER_URL_PARAMETER,
 } from '../config'
 import { replaceComponent } from '../routeComponentLifeCycle'
 import store from '../store'
 import AfterOauthLogin from '../components/screens/AfterOauthLogin.svelte'
-import {
-  fetchCurrentUserRepositories,
-  createRepositoryForCurrentAccount,
-} from '../actions.js'
+import { fetchCurrentUserRepositories } from '../actions/current-user.js'
+import { createRepositoryForCurrentAccount } from '../actions/setup.js'
+import { oAuthAppByType } from '../oauth-services-api/index.js'
 
-const storeOAuthProviderAccess = () => {
+const storeOAuthProviderAccess = async () => {
   const url = new URL(location.href)
 
   console.log(
     'type',
     url.searchParams.get(TOCTOCTOC_OAUTH_PROVIDER_URL_PARAMETER),
   )
 
-  if (
-    url.searchParams.has(TOCTOCTOC_ACCESS_TOKEN_URL_PARAMETER) &&
-    url.searchParams.get(TOCTOCTOC_ACCESS_TOKEN_URL_PARAMETER) !== null &&
-    url.searchParams.has(TOCTOCTOC_OAUTH_PROVIDER_URL_PARAMETER) &&
-    url.searchParams.get(TOCTOCTOC_OAUTH_PROVIDER_URL_PARAMETER) !== null
-  ) {
-    const accessToken = url.searchParams.get(
-      TOCTOCTOC_ACCESS_TOKEN_URL_PARAMETER,
-    )
-    const providerName = url.searchParams.get(
-      TOCTOCTOC_OAUTH_PROVIDER_URL_PARAMETER,
-    )
-
-    // @ts-ignore
-    localStorage.setItem(ACCESS_TOKEN_STORAGE_KEY, accessToken)
-    // @ts-ignore
-    localStorage.setItem(OAUTH_PROVIDER_STORAGE_KEY, providerName)
-
-    store.mutations.setOAuthProvider({
-      accessToken: accessToken,
+  const accessToken = url.searchParams.get(TOCTOCTOC_ACCESS_TOKEN_URL_PARAMETER)
+  const providerName = url.searchParams.get(
+    TOCTOCTOC_OAUTH_PROVIDER_URL_PARAMETER,
+  )
+
+  if (accessToken && providerName) {
+    const origin = oAuthAppByType.get(providerName)?.origin || ''
+    const oAuthProvider = {
       name: providerName,
-    })
+      accessToken,
+      origin,
+    }
+
+    await remember(OAUTH_PROVIDER_STORAGE_KEY, oAuthProvider)
+
+    store.mutations.setOAuthProvider(Promise.resolve(oAuthProvider))
```

Comment:
```suggestion
    store.mutations.setOAuthProvider(oAuthProvider)
```

On a l'objet direct, donc pas besoin de le wrap dans une promesse
Ça va permettre de l'utiliser dans les autres fonctions qui utilisent `store.state.oAuthProvider`